src/sagemaker/jumpstart/enums.py

```diff
@@ -126,3 +128,88 @@ def from_suffixed_type(mime_type_with_suffix: str) -> "MIMEType":
         """Removes suffix from type and instantiates enum."""
         base_type, _, _ = mime_type_with_suffix.partition(";")
         return MIMEType(base_type)
+
+
+class NamingConventionType(str, Enum):
+    """Enum class for naming conventions."""
+
+    SNAKE_CASE = "snake_case"
+    UPPER_CAMEL_CASE = "upper_camel_case"
+    DEFAULT = UPPER_CAMEL_CASE
+
+    @staticmethod
+    def upper_camel_to_snake(upper_camel_case_string: str) -> str:
+        """Converts UpperCamelCaseString to snake_case_string."""
+        snake_case_string = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", upper_camel_case_string)
+        return re.sub("([a-z0-9])([A-Z])", r"\1_\2", snake_case_string).lower()
+
+    @staticmethod
+    def snake_to_upper_camel(snake_case_string: str) -> str:
+        """Converts snake_case_string to UpperCamelCaseString."""
+        upper_camel_case_string = "".join(word.title() for word in snake_case_string.split("_"))
+        return upper_camel_case_string
+
+
+class ModelSpecKwargType(str, Enum):
+    """Enum class for types of kwargs for model hub content document and model specs."""
+
+    FIT = "fit_kwargs"
+    MODEL = "model_kwargs"
+    ESTIMATOR = "estimator_kwargs"
+    DEPLOY = "deploy_kwargs"
+
+    @classmethod
+    def arg_keys(cls) -> List[str]:
+        """Returns a list of kwargs keys that each type can have"""
+        return [member.value for member in cls]
+
+    @staticmethod
+    def get_model_spec_arg_keys(
+        arg_type: "ModelSpecKwargType",
```

Comment:
i think this typing convention may not work on older py3 versions

Reply:
HHmm let me check.